### DISCUSSION TITLE
Fortigate Integration Issues

### DISCUSSION THREAD

**Discussion by jaypfe** [Security-Onion-Solutions/securityonion#11504] [2.4]
### Version

2.4.20

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

24

### Storage for /

256

### Storage for /nsm

256

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

I am having issues getting the FortiGate module to work at all. I am not seeing any kind of data stream being created period. 
Please see below for the config 

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/2113b779-2d40-47dc-9557-7357bb0e933a)
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/af325c4a-28d0-4986-a5f2-d9cf63d04e85)
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/e924b195-850c-401d-809b-3a6acbaa3954)
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/fd04dbdd-e52f-49c8-a3f0-f70c7b20686d)

I added it to syslog just to check my sanity and I can get logs that way, so I know the firewall is sending them. 
Within the integration config: 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/5f0ecb27-4f9a-470d-8653-0d2f724a62b8)
I used 0.0.0.0 to bind all interfaces, I applied it to both the grid nodes general and fleet server polices with no difference of outcome. 

I dont even see a data stream being created here, pretty weird: 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/c5a6ecd4-e39c-4d8c-abe4-977d8f5d6a07)

I made another config, as we did in .10 but still resulted in no data stream. 
looking in /opt/so/log under Elasticsearch I am seeing a buuuuunch of: 
[2023-10-10T18:08:48,679][WARN ][org.elasticsearch.ingest.common.GrokProcessor] character class has '-' without escape

I am not sure if that means anything significant. There are a couple of other WARNS in there. I can dump the file if wanted. 

Looking with netstat it does look like the port is open: 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/32fb0a7f-6196-4354-bc0b-5887541e47f9)

I just dont see any traffic through it looking at with tcpdump other than 1 packet with a length of 0 every once and a while. 

Please let me know if I can provide anything more to help with this. 

Loved the conference and the great work you guys do!

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by keis3cker** [comment]
Hi,
As you are using tcp to collect the data, try to set;  framing: rfc6587 under advanced options in the fortigate integration settings.

**Reply by jaypfe** [reply]
Thanks for the suggestion, unfortunately however it does not appear to have changed. 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/25958a22-3e01-46d2-ab53-a8db628815be)

**Reply by keis3cker** [reply]
Looking at your netstat screenshot. It seems like the port is only listening on loopback 127.0.0.1 instead of 0.0.0.0 

I would try to remove all configured fortigate integrations from the agent-policies and then just configure it for 1 policy (for example grid-nodes policy). In normal case all gridnodes would expose this port.

Then review the firewall configuration. You should be able to test if the port is open by using tools like nmap.

**Comment by jaypfe** [comment]
@keis3cker  I had something weird going on with the IP tables, it never allowed the host group. Changing that, I get traffic in now. Thanks for sending me down that path!

buuut I am having the same issue I was in .10. A data stream gets built, the correct template gets selected, I can see EPS in influx, but nothing gets written into the data stream. 

If I change the index.final.pipeline value on the index from fleet.final to the just to FortiGate pipeline, it starts writing to the index. I have no idea what would cause this behavior. Maybe fleet final does not like it? Something in the FortiGate pipeline is making life hard for fleet final?

**Reply by jaypfe** [reply]
Using the depricated fortinet integration, everything works like you would expect. Almost has to be an issue with the fortigate integration now.

**Reply by keis3cker** [reply]
Ingest piplines can be a mess sometimes and debugging is hard. I have a standalone ELK cluster for collecting fortigate logs and their ingest piplines had alot of issues at the beginning. The latest version of the integration is 1.19.0 which finally runs fine for me. The version shipped with SO (airgap)  is 1.14.0, maybe this version still has some bugs in their ingest piplines.
